### PR TITLE
Persist implementation

### PR DESCRIPTION
**Pull request by syevtushok**

### CODE REVIEW

File: src/main/java/com/breskul/bibernate/annotation/Entity.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.breskul.bibernate.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bibernate/annotation/Entity.java:7]: We should add java doc to public methods, annotations, classes, enums, exception classes and public util methods too.

**Review by yuriyfomin17 — APPROVED** [review]

File: src/main/java/com/breskul/bibernate/persistence/EntityManagerImpl.java [hunk 2]
```diff
@@ -1,29 +1,106 @@
...
+        List<Object> values = new ArrayList<>();
+        Queue<Collection<?>> queue = new ArrayDeque<>();
+        for (Field declaredField : entityToSave.getClass().getDeclaredFields()) {
+            if (declaredField.isAnnotationPresent(Id.class)) {
+                Field idField = getFieldByAnnotation(entityToSave.getClass(), Id.class);
+                Object entityId = jdbcDao.resolveEntityId(entityToSave, idField);
+                if (entityId != null) {
+                    processRegularField(columns, values, idField, entityId);
+                }
+                continue;
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bibernate/persistence/EntityManagerImpl.java:59]: I think better use else construction and avoid continue usage
image

File: src/main/java/com/breskul/bibernate/persistence/util/DaoUtils.java [hunk 3]
```diff
@@ -42,4 +52,55 @@ public static String getClassTableName(Class<?> entityClass) {
                 .orElse(entityClass.getSimpleName());
 
     }
+
+    public static <T> void isValidEntity(Class<T> type) {
+        if (type.isAnnotationPresent(Entity.class)) {
```
Comment (Mikhail-Mikhailenko) [inline src/main/java/com/breskul/bibernate/persistence/util/DaoUtils.java:57]: Could you please add tests for utils methods

**Review by Mikhail-Mikhailenko — COMMENTED** [review]
Could you please add documantation for for your feature inside Readme file.

**Comment by yuriyfomin17** [comment]
merging as discussed with Serhii and Misha to avoid future conflicts. Added trello tasks to add documentation and tests

**Pull request merged**